R/stan_clogit.R

```diff
@@ -0,0 +1,174 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+#' Conditional logistic (clogit) models via Stan
+#'
+#' A model for case-control studies with optional prior distributions for the
+#' coefficients, intercept, and auxiliary parameters.
+#'
+#' @export
+#' @templateVar pkg survival
+#' @templateVar pkgfun clogit
+#' @templateVar sameargs model,offset
+#' @templateVar rareargs na.action,contrasts
+#' @templateVar fun stan_clogit
+#' @templateVar fitfun stan_glm.fit
+#' @template return-stanreg-object
+#' @template see-also
+#' @template args-priors
+#' @template args-prior_PD
+#' @template args-algorithm
+#' @template args-adapt_delta
+#' @template args-QR
+#' @template args-sparse
+#' @template args-dots
+#' 
+#' @param formula,data,subset,na.action Same as for \code{\link[lme4]{glmer}}, 
+#'   except that any intercept included in the formula will be dropped. \emph{We
+#'   strongly advise against omitting the \code{data} argument}. Unless 
+#'   \code{data} is specified (and is a data frame) many post-estimation 
+#'   functions (including \code{update}, \code{loo}, \code{kfold}) are not 
+#'   guaranteed to work properly.
+#' @param strata A factor indicating the groups in the data where the number of 
+#'   successes (possibly one) is fixed by the research design. It may be useful 
+#'   to use \code{\link{interaction}} or \code{\link[survival]{strata}} to
+#'   create this factor.
+#' @param prior_covariance Cannot be \code{NULL} when lme4-style group-specific
+#'   terms are included in the \code{formula}. See \code{\link{decov}} for
+#'   more information about the default arguments. Ignored when there are no
+#'   group-specific terms.
+#' 
+#' @details The \code{stan_clogit} function is mostly similar in syntax to 
+#'   \code{\link[survival]{clogit}} but rather than performing maximum
+#'   likelihood estimation of generalized linear models, full Bayesian
+#'   estimation is performed (if \code{algorithm} is \code{"sampling"}) via
+#'   MCMC. The Bayesian model adds priors (independent by default) on the
+#'   coefficients of the GLM.
+#'   
+#'   The \code{data.frame} passed to the \code{data} argument must be sorted by 
+#'   the variable passed to the \code{strata} argument.
+#'   
+#'   The \code{formula} may have group-specific terms like in 
+#'   \code{\link{stan_glmer}} but should not allow the intercept to vary by the
+#'   stratifying variable, since there is no information in the data with which
+#'   to estimate such deviations in the intercept.
+#'   
+#' @seealso The vignette for Bernoulli and binomial models.
+#' 
+#' @examples
```

Comment:
If lme4 syntax is allowed I think there should be an example

Reply:
I am adding `(1 | education)` to the example.